apps/emails-and-messages/src/pages/api/webhooks/notify.ts

```diff
@@ -0,0 +1,122 @@
+import { NextWebhookApiHandler, SaleorAsyncWebhook } from "@saleor/app-sdk/handlers/next";
+import { gql } from "urql";
+import { saleorApp } from "../../../saleor-app";
+import { logger as pinoLogger } from "../../../lib/logger";
+import { sendEventMessages } from "../../../modules/event-handlers/send-event-messages";
+import { createClient } from "../../../lib/create-graphql-client";
+import { MessageEventTypes } from "../../../modules/event-handlers/message-event-types";
+
+// Notify event handles multiple event types which are recognized b
+
+export interface NotifyPayload {
+  notify_event: string;
+  payload: EventPayload;
+  meta: Meta;
+}
+
+export interface Meta {
+  issued_at: Date;
+  version: string;
+  issuing_principal: IssuingPrincipal;
+}
+
+export interface IssuingPrincipal {
+  id: null | string;
+  type: null | string;
+}
+
+export interface EventPayload {
+  user: User;
+  recipient_email: string;
+  channel_slug: string;
+  domain: string;
+  site_name: string;
+  logo_url: string;
+  token?: string;
+  confirm_url?: string;
+  reset_url?: string;
+  delete_url?: string;
+  old_email?: string;
+  new_email?: string;
+  redirect_url?: string;
+}
+
+export interface User {
+  id: string;
+  email: string;
+  first_name: string;
+  last_name: string;
+  is_staff: boolean;
+  is_active: boolean;
+  private_metadata: Metadata;
+  metadata: Metadata;
+  language_code: string;
+}
+
+export interface Metadata {}
+
+export const notifyWebhook = new SaleorAsyncWebhook<NotifyPayload>({
+  name: "notify",
+  webhookPath: "api/webhooks/notify",
+  asyncEvent: "NOTIFY_USER",
+  apl: saleorApp.apl,
+  query: "{}", // We are using the default payload instead of subscription
+});
+
+const handler: NextWebhookApiHandler<NotifyPayload> = async (req, res, context) => {
+  const logger = pinoLogger.child({
+    webhook: notifyWebhook.name,
+  });
+
+  logger.debug("Webhook received");
+
+  const { payload, authData } = context;
+
+  const { channel_slug: channel, recipient_email: recipientEmail } = payload.payload;
+
+  if (!recipientEmail?.length) {
+    logger.error(`The email recipient has not been specified in the event payload.`);
+    return res
+      .status(200)
+      .json({ error: "Email recipient has not been specified in the event payload." });
+  }
+
+  const notifyEventMapping: Record<string, MessageEventTypes> = {
+    account_confirmation: "ACCOUNT_CONFIRMATION",
+    account_delete: "ACCOUNT_DELETE",
+    account_password_reset: "ACCOUNT_PASSWORD_RESET",
+    account_change_email_request: "ACCOUNT_CHANGE_EMAIL_REQUEST",
+    account_change_email_confirm: "ACCOUNT_CHANGE_EMAIL_CONFIRM",
+  };
```

Comment:
This is important fragment - sort of "adapter" from legacy webhook to new ones. I don't think we need to build abstraction, but I'd like to see some comment that will describe the context (that other events are missing so its mapped here, but new events will be added in time)

Reply:
Added context comment explaining why it has to be mapped in this webhook